BitwardenShared/Core/Platform/Services/StateService.swift

```diff
@@ -1494,11 +1494,19 @@ actor DefaultStateService: StateService { // swiftlint:disable:this type_body_le
 
     func getVaultTimeout(userId: String?) async throws -> SessionTimeoutValue {
         let userId = try getAccount(userId: userId).profile.userId
-        guard let rawValue = appSettingsStore.vaultTimeout(userId: userId) else {
-            let userAuthKey = try? await keychainRepository.getUserAuthKeyValue(for: .neverLock(userId: userId))
-            return userAuthKey == nil ? .fifteenMinutes : .never
+        let userAuthKey = try? await keychainRepository.getUserAuthKeyValue(for: .neverLock(userId: userId))
+        let timeoutValue: SessionTimeoutValue = if let rawValue = appSettingsStore.vaultTimeout(userId: userId) {
+            SessionTimeoutValue(rawValue: rawValue)
+        } else {
+            // If there isn't a stored value, it may be because MAUI stored `nil` for never timeout.
+            // So if the never lock key exists, set the timeout to never.
+            userAuthKey != nil ? .never : .fifteenMinutes
+        }
+        if timeoutValue == .never, userAuthKey == nil {
+            // If never lock but no key (possibly due to logging out), return the default timeout.
+            return .fifteenMinutes
         }
-        return SessionTimeoutValue(rawValue: rawValue)
+        return timeoutValue
```

Comment:
🎨 🤔 I'll leave it up to you but I think this could be written in the next way to reduce cyclomatic complexity and be easier to folllow:
```swift
let userAuthKey = try? await keychainRepository.getUserAuthKeyValue(for: .neverLock(userId: userId))
guard let rawValue = appSettingsStore.vaultTimeout(userId: userId) else {
    // If there isn't a stored value, it may be because MAUI stored `nil` for never timeout.
    // So if the never lock key exists, set the timeout to never, otherwise to default.
    return userAuthKey != nil ? .never : .fifteenMinutes
}

let timeoutValue = SessionTimeoutValue(rawValue: rawValue)
if timeoutValue == .never, userAuthKey == nil {
    // If never lock but no key (possibly due to logging out), return the default timeout.
    return .fifteenMinutes
}
return timeoutValue
```
Given that `userAuthKey != nil ? .never : .fifteenMinutes` will cause the `if timeoutValue == .never, userAuthKey == nil` to never apply so we can return early in that case.
What do you think?

Reply:
I like that!